rebasehelper/cli.py

```diff
@@ -103,8 +104,32 @@ def __getattr__(self, name):
 
 class CliHelper:
 
-    @staticmethod
-    def run():
+    @classmethod
+    def convert_macros_to_dict(cls, macros_list: List[str]) -> Dict[str, str]:
+        """Converts macros from CLI to a dictionary.
+
+        Args:
+            macros_list: List of macros in the format 'MACRO EXPRESSION'.
+
+        Returns:
+            The converted macros, MACRO are keys, EXPRESSION are values.
+
+        Raises:
+            RebaseHelperError if the macros don't follow the correct format.
+
+        """
+        macros = {}
+        for macro in macros_list:
+            match = re.match(r"\s*%?(?P<macro>\S+)\s+(?P<expr>\S+(\s*\S+)*)", macro)
```

Comment:
Why is `\S+(\s*\S+)*` necessary?

Reply:
There are a few reasons - `.*` can't be used instead, because something like `macro    ` would match but it shouldn't so `\S+` has to be used instead. However, `rpm -D` succeeds even with `macro something1 something2`, which wouldn't work with just single '\S+' (only the first word would be taken). I am not completely positive how rpm handles such case, I assume that it takes the first word as macro name and the rest as the value but I don't know how to confirm it. So I added the optional `\S+` separated with any number of whitespaces.

Reply:
How would `macro `  match? There is `\s+`.

Reply:
Well, I tested it and it seems that with 2 spaces, `.+` takes priority over `\s+` and the regex isn't evaluated from left to right. See [this as an example](https://pythex.org/?regex=%5Cs*%25%3F(%3FP%3Cmacro%3E%5CS%2B)%5Cs%2B(%3FP%3Cexpr%3E.%2B)&test_string=%20macro%20%20%20&ignorecase=1&multiline=0&dotall=0&verbose=0)

Reply:
No, that's not what happens, `.+` matches the last space (because of the `+`). How about this?
`\s*%?(?P<macro>\S+)\s+(?P<expr>\S.*)\s*`